scalafmt-core/shared/src/main/scala/org/scalafmt/util/StyleMap.scala

```diff
@@ -27,7 +27,9 @@ class StyleMap(
   import TokenOps.hash
   val literalR: FilterMatcher = init.binPack.literalsRegex
   private val prefix = "\\s*scalafmt: ".r
-  val forcedBinPack: mutable.Set[Tree] = mutable.Set.empty
+  private val forcedBinPackk = mutable.ArrayBuffer.empty[Tree]
+
+  def forcedBinPack(t: Tree) = forcedBinPackk.exists(_.eq(t)) //only check the reference instead of hashing the tree completely has high improvement and the tree should be unique in memory
```

Comment:
FYI, trees already use reference equality for `hashCode` and `equals`  https://github.com/scalameta/scalameta/blob/dc639c83f1c26627c39aef9bfb3dae779ecdb237/scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala#L17-L19

If you are noticing performance improvements, it could be because we avoid an `invokeinterface` call on the `Tree` trait